### PR TITLE
created chat table structure and added messagetypes

### PR DESCRIPTION
**Pull request by navjotcis**
1) Create a chat table structure
2) added messagetypes in message directory
3) write a separate test case for decode and encode method

### CODE REVIEW

File: src/class_sqlThread.py [hunk 1]
```diff
@@ -392,6 +392,22 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
                 ' and removing the hash field.')
             self.cur.execute('''update settings set value=10 WHERE key='version';''')
 
+        # Add a new table: chat and chatdata for storing chating conversation
+        item = '''SELECT value FROM settings WHERE key='version';'''
+        parameters = ''
+        self.cur.execute(item, parameters)
+        currentVersion = int(self.cur.fetchall()[0][0])
+        if currentVersion == 10:
+            self.cur.execute('''DROP TABLE chat''')
```
Comment (PeterSurda) [inline src/class_sqlThread.py:401]: DROP TABLE IF EXISTS

File: src/class_sqlThread.py [hunk 2]
```diff
@@ -392,6 +392,22 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
...
+            self.cur.execute(
+                '''CREATE TABLE chat'''
+                ''' (msgid blob, senderaddress text, receiveraddress text, message text,'''
+                ''' senttime text, receivedtime text, image blob, audio blob, reference blob,'''
```
Comment (PeterSurda) [inline src/class_sqlThread.py:405]: no senttime

File: src/messagetypes/__init__.py [hunk 3]
```diff
@@ -20,7 +20,7 @@ def __init__(self):
 
 def constructObject(data):
     """Constructing an object"""
-    whitelist = ["message"]
+    whitelist = ["message", "chatmsg"]
```
Comment (PeterSurda) [inline src/messagetypes/__init__.py:23]: maybe "chat" rather than "chatmsg"

File: src/class_sqlThread.py (hunk 1 above)
Comment (navjotcis) [reply src/class_sqlThread.py:401]: yes I have already removed it in next push

**Pull request merged**